packages/llvm.rb

```diff
@@ -1,276 +1,104 @@
-       i686: '8c4173f7de0525948ce60fed8d7fef7a3f1cb4632b650e6623eed6b4a0db7cce',
-     x86_64: 'b4751906ff72b3e4e2fcfbf3380bdc17907e3463c86ba059113369c459c7783f',
+     aarch64: '5ecb72e60483b15c7588446a7ac424d64cdd3f21ca0fa5ffea4737a25dedca89',
+      armv7l: '5ecb72e60483b15c7588446a7ac424d64cdd3f21ca0fa5ffea4737a25dedca89',
+        i686: '5595ca097aa454bd54b8a336d467ee266f8afd4bd8100d1a3831e3face9b4f43',
+      x86_64: '031f6f3510ebbc5a1270fac07934d7efa9126c8e89f98695505b1376d6d233a5',
   })
+  
+  # llvm_stage1 is compiled with gcc, without -flto
+  # Then in stage 2 (llvm.rb) everything is compiled with clang built in stage 1, with -flto=thin
+  # FYI LLVM 11.1.0 is a tiny rebuild of 11.0.1 fixing ABI compatibility with LLVM 10.0.0 & LLVM 12
 
-  depends_on 'ld_default' => :build
-  depends_on 'graphviz' => :build # for docs
-  depends_on 'python27' => :build # for test suite
-  depends_on 'sphinx' => :build
   depends_on 'ocaml' => :build
   depends_on 'libedit'
   depends_on 'libtirpc'
   depends_on 'swig'
+  depends_on 'pygments' => :build
+  depends_on 'ccache' => :build
+  depends_on 'llvm_stage1' => :build
+  depends_on 'binutils' => :build
+
+  case ARCH
+  when 'aarch64','armv7l'
+    LLVM_DEFAULT_TARGET_TRIPLE = "armv7l-cros-linux-gnueabihf"
+    #LLVM_TARGETS_TO_BUILD = 'ARM;AArch64;AMDGPU'
+    LLVM_TARGETS_TO_BUILD = 'all'
+    @ARCH_C_FLAGS = "-fPIC -march=armv7-a -mfloat-abi=hard -ccc-gcc-name #{LLVM_DEFAULT_TARGET_TRIPLE}"
+    @ARCH_CXX_FLAGS = "-fPIC -march=armv7-a -mfloat-abi=hard -ccc-gcc-name #{LLVM_DEFAULT_TARGET_TRIPLE}"
+    @ARCH_LDFLAGS=''
+  when 'i686'
+    LLVM_DEFAULT_TARGET_TRIPLE = "#{ARCH}-cros-linux-gnu"
+    LLVM_TARGETS_TO_BUILD = 'X86'
+    # Because ld.lld: error: undefined symbol: __atomic_store
+    # Polly demands fPIC
+    @ARCH_C_FLAGS = '-latomic -fPIC'
+    @ARCH_CXX_FLAGS = '-latomic -fPIC'
+    # Because getting this error:
+    # ld.lld: error: relocation R_386_PC32 cannot be used against symbol isl_map_fix_si; recompile with -fPIC
+    # So as per https://github.com/openssl/openssl/issues/11305#issuecomment-602003528
+    @ARCH_LDFLAGS='-Wl,-znotext'
+  when 'x86_64'
+    LLVM_DEFAULT_TARGET_TRIPLE = "#{ARCH}-cros-linux-gnu"
+    #LLVM_TARGETS_TO_BUILD = 'X86;AMDGPU'
+    LLVM_TARGETS_TO_BUILD = 'all'
+    @ARCH_C_FLAGS = '-fPIC'
+    @ARCH_CXX_FLAGS = '-fPIC'
+    @ARCH_LDFLAGS=''
+  end
+  @ARCH_C_LTO_FLAGS = "#{@ARCH_C_FLAGS} -flto=thin -fuse-ld=lld"
+  @ARCH_CXX_LTO_FLAGS = "#{@ARCH_CXX_FLAGS} -flto=thin -fuse-ld=lld"
+  LLVM_PROJECTS_TO_BUILD = 'clang;clang-tools-extra;libcxx;libcxxabi;libunwind;lldb;compiler-rt;lld;polly;openmp'
+  
+  # Using Targets 'all' for non-i686 because otherwise mesa complains.
+  # This may be patched upstream as per 
+  # https://reviews.llvm.org/rG1de56d6d13c083c996dfd44a32041dacae037d66
+  #LLVM_TARGETS_TO_BUILD = 'all'
+  LLVM_VERSION = version.split("-")[0]
+  BINUTILS_BRANCH = 'gdb-10.1-release'
```

Comment:
I don't see these constants used anywhere.  Can we remove?

Reply:
Removed.